packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -110,57 +144,195 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
       path: driftStoragePath,
     );
 
-    __analyticsClient = await AnalyticsClient.getInstance(
+    _eventCreator = EventCreator(
+      globalFieldsManager: _eventGlobalFieldsManager,
+      deviceContextInfo: deviceContextInfo,
+    );
+
+    // Retrieve Unique ID
+    final savedFixedEndpointId =
+        await _keyValueStore.read(key: endpointIdStorageKey);
+    final fixedEndpointId = savedFixedEndpointId ?? const Uuid().v1();
+    if (savedFixedEndpointId == null) {
+      await _keyValueStore.write(
+        key: endpointIdStorageKey,
+        value: fixedEndpointId,
+      );
+    }
+
+    final endpoint = PublicEndpoint(
+      effectiveDate: DateTime.now().toUtc().toIso8601String(),
+      demographic: EndpointDemographic(
+        appVersion: deviceContextInfo?.appVersion,
+        locale: deviceContextInfo?.locale,
+        make: deviceContextInfo?.make,
+        model: deviceContextInfo?.model,
+        modelVersion: deviceContextInfo?.modelVersion,
+        platform: deviceContextInfo?.platform?.name,
+        platformVersion: deviceContextInfo?.platformVersion,
+        timezone: deviceContextInfo?.timezone,
+      ),
+      location: EndpointLocation(
+        country: deviceContextInfo?.countryCode,
+      ),
+    );
+
+    /// Retrieve stored GlobalAttributes
+    final cachedAttributes =
+        await _keyValueStore.read(key: _endpointGlobalAttrsKey);
+    final globalAttributes = cachedAttributes == null
+        ? <String, String>{}
+        : (jsonDecode(cachedAttributes) as Map<String, Object?>)
+            .cast<String, String>();
+
+    /// Retrieve stored GlobalMetrics
+    final cachedMetrics =
+        await _keyValueStore.read(key: _endpointGlobalMetricsKey);
+    final globalMetrics = cachedMetrics == null
+        ? <String, double>{}
+        : (jsonDecode(cachedMetrics) as Map<String, Object?>)
+            .cast<String, double>();
+
+    _endpointGlobalFieldsManager = EndpointGlobalFieldsManager(
+      _keyValueStore,
+      globalAttributes,
+      globalMetrics,
+    );
+
+    _endpointClient = EndpointClient(
+      appId,
+      fixedEndpointId,
+      pinpointClient,
+      _endpointGlobalFieldsManager,
+      endpoint.toBuilder(),
+    );
+    unawaited(
+      Amplify.asyncConfig.then((_) {
+        _endpointClient.updateEndpoint();
+      }),
+    );
+
+    _eventStorageAdapter = EventStorageAdapter(driftQueryExecutor);
+
+    _eventClient = EventClient(
       appId: appId,
-      keyValueStore: keyValueStore,
+      fixedEndpointId: fixedEndpointId,
       pinpointClient: pinpointClient,
-      driftQueryExecutor: driftQueryExecutor,
+      endpointClient: _endpointClient,
+      storageAdapter: _eventStorageAdapter,
+    );
+
+    _sessionManager = SessionManager(
```

Comment:
That is the current target behavior.  iOS and Android do not serialize the isEnabled property for Analytics either.  Doing so would cause our behavior to differ.  I can check with those teams and with Abdallah about that behavior too.